irc/prometheus.go

```diff
@@ -0,0 +1,32 @@
+package irc
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	command_received = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "command_received_total",
+			Help: "Number of irc commands received.",
+		},
+		[]string{"nick", "command"},
+	)
+)
+
+func init() {
```

Comment:
Can these be defined/registered where they are used?

Reply:
Probably. Every metric needs to be defined once before it is used, I've only ever seen them defined together under a prometheus namespace and then initialized at bootstrap in any programming language. The go examples I found used an init function (which I guess is automatically run) like this.

I think they are primarily defined together to make finding all the metrics easier, and because a metric might be used in multiple places throughout a codebase.